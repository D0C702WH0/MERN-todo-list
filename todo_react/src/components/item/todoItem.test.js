import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import TodoItem from './todoItem';

Enzyme.configure({ adapter: new Adapter() });

let item;
let todoRemove;
let todoUpdate;
const todo = {
  _id: "kmkmlk",
  title: 'test',
  done: false,
  isActive: true
}

beforeEach(() => {
  todoRemove = jest.fn()
  todoUpdate = jest.fn()
  item = shallow(
    <TodoItem
      key={todo._id}
      data={todo}
      todoRemove={todoRemove}
      todoUpdate={todoUpdate}
    />);
});

describe('Input', () => {

  it('Should exist', () => {
    expect(item).toBeDefined();
    expect(item.exists()).toBe(true);
  });

  test('it matches snapshot', () => {
    const itemSnap = renderer.create(<TodoItem
      key={todo._id}
      data={todo}
      todoRemove={todoRemove}
      todoUpdate={todoUpdate}
    />).toJSON();
    expect(itemSnap).toMatchSnapshot();
  });

  it('Should call todoRemove() on btn click', () => {
    const btn = item.find('button');
    const todoRemove = item.instance().props.todoRemove;
    btn.simulate('click');
    expect(todoRemove).toHaveBeenCalled();
  });

  it('Should call todoUpdate() on box checked', () => {
    const box = item.find('input');
    const todoUpdate = item.instance().props.todoUpdate;
    box.simulate('change');
    expect(todoUpdate).toHaveBeenCalled();
  });

  it('Should empty span className on box unchecked', () => {
    const span = item.find('span')
    item.instance().state.checked = false
    expect(span.hasClass('done')).toEqual(false);
  });
})
