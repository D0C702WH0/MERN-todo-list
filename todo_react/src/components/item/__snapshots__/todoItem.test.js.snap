// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input it matches snapshot 1`] = `
<div
  className="card col-md-3 m-2"
>
  <div
    className="card-body"
  >
    <div
      className="dataContainer"
    >
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className=""
      >
        test
      </span>
      <button
        className="btn btn-danger"
        onClick={[Function]}
        type="button "
      >
        X
      </button>
    </div>
  </div>
</div>
`;
